library/vyos_interfaces.py

```diff
@@ -0,0 +1,396 @@
+# set interfaces ethernet eth3 description 'Configured by Ansible Network'
+# set interfaces ethernet eth3 duplex 'full'
+# set interfaces ethernet eth3 hw-id '08:00:27:17:3c:85'
+# set interfaces ethernet eth3 smp_affinity 'auto'
+# set interfaces ethernet eth3 speed '100'
+
+- name: Overrides all device configuration with provided configuration
+  vyos_interfaces:
+    config:
+      - name: eth1
+        description: Outbound Interface For The Appliance
+
+      - name: eth2
+        speed: auto
+        duplex: auto
+
+      - name: eth3
+        mtu: 1200
+    state: overridden
+
+# After state
+# ------------
+#
+# vyos@vyos-appliance:~$ show configuration commands | grep eth
+# set interfaces ethernet eth0 address 'dhcp'
+# set interfaces ethernet eth0 hw-id '08:00:27:f3:6c:b5'
+# set interfaces ethernet eth0 smp_affinity 'auto'
+# set interfaces ethernet eth1 address '192.168.1.120/24'
+# set interfaces ethernet eth1 description 'Outbound Interface For The Appliance'
+# set interfaces ethernet eth1 hw-id '08:00:27:ad:ef:65'
+# set interfaces ethernet eth1 smp_affinity 'auto'
+# set interfaces ethernet eth2 duplex 'auto'
+# set interfaces ethernet eth2 hw-id '08:00:27:ab:4e:79'
+# set interfaces ethernet eth2 smp_affinity 'auto'
+# set interfaces ethernet eth2 speed 'auto'
+# set interfaces ethernet eth3 hw-id '08:00:27:17:3c:85'
+# set interfaces ethernet eth3 mtu '1200'
+# set interfaces ethernet eth3 smp_affinity 'auto'
+
+
+# Using deleted
+#
+# Before state
+# -------------
+# vyos@vyos-appliance:~$ show configuration commands | grep eth
+# set interfaces ethernet eth0 address 'dhcp'
+# set interfaces ethernet eth0 description 'Outbound Interface for this appliance'
+# set interfaces ethernet eth0 duplex 'auto'
+# set interfaces ethernet eth0 hw-id '08:00:27:f3:6c:b5'
+# set interfaces ethernet eth0 smp_affinity 'auto'
+# set interfaces ethernet eth0 speed 'auto'
+# set interfaces ethernet eth1 description 'Configured by Ansible Network'
+# set interfaces ethernet eth1 duplex 'full'
+# set interfaces ethernet eth1 hw-id '08:00:27:ad:ef:65'
+# set interfaces ethernet eth1 smp_affinity 'auto'
+# set interfaces ethernet eth1 speed '100'
+# set interfaces ethernet eth2 description 'Configured by Ansible'
+# set interfaces ethernet eth2 duplex 'full'
+# set interfaces ethernet eth2 hw-id '08:00:27:ab:4e:79'
+# set interfaces ethernet eth2 mtu '600'
+# set interfaces ethernet eth2 smp_affinity 'auto'
+# set interfaces ethernet eth2 speed '100'
+# set interfaces ethernet eth3 description 'Configured by Ansible Network'
+# set interfaces ethernet eth3 duplex 'full'
+# set interfaces ethernet eth3 hw-id '08:00:27:17:3c:85'
+# set interfaces ethernet eth3 smp_affinity 'auto'
+# set interfaces ethernet eth3 speed '100'
+
+- name: Delete attributes of given interfaces (Note: This won't delete the interface itself)
+  vyos_interfaces:
+    config:
+      - name: "{{ item }}"
+    state: deleted
+    loop:
```

Comment:
We don't need to use loop here since the module config should take a list of interfaces to delete

Reply:
Solved.